sink/src/main/kotlin/org/neo4j/connectors/kafka/sink/SinkStrategy.kt

```diff
@@ -164,6 +165,8 @@ interface SinkStrategyHandler {
                 bindHeaderAs = config.patternBindHeaderAs,
                 bindKeyAs = config.patternBindKeyAs,
                 bindValueAs = config.patternBindValueAs)
+
+        handler.validate(fetchConstraintData(config.driver.session(config.sessionConfig())))
```

Comment:
maybe a future enhancement***: the PR currently checks the constraints only once per handler instantiation, right? constraints can change and the pattern which could be valid once could then become invalid. Maybe we need to run this validation on a regular basis instead (probably not for this PR though, that'll likely complicate this)



***or not: I'm not clear whether this is good idea or not